types/src/transaction/analyzed_transaction.rs

```diff
@@ -131,7 +131,7 @@ impl AnalyzedTransaction {
         AnalyzedTransaction::new(
             Transaction::UserTransaction(signed_txn),
             // Please note that we omit all the modules we read and the global supply we write to?
-            vec![],
```

Comment:
@sitalkedia here you meant read-only storage locations so it has to be left empty, right?

Reply:
Yes, @andrekorol - Any reason to change this?

Reply:
If it is empty, we have no read-write conflicts and write-write conflicts aren't really conflicts when we have multiversioning. So, it gave, essentially, conflict-free workloads.

Reply:
write-write conflict is also treated as read-write conflict as the Move VM issues a read before all writes.

Reply:
Then I guess it is could be reasonable (and more convenient) to keep `write_set` a subset of `read_set`? To avoid doing it manually in the partitioner itself.

Reply:
Can you explain why its more convenient? Unless we figure out the exclusive read set, the partitioner will do unnecessary extra work in finding read set conflicts.

Reply:
Not sure what you mean by "exclusive read set" and "read set conflicts". The "more convenient" part may depend on the specific partitioner implementation. In the case of this prototype, it needs to detect "read-write" conflicts and it's easiest to do when you actually have your full read set available via `read_set()`. Otherwise, it would need to manually iterate over both `read_set()` and `write_set()` to get the actual read set (since, as you mentioned, a transaction that writes to a location also reads that location). If you want, I can keep this shared code intact and just add the write set to the read set when passing it to this partitioner.

Reply:
I would definitely make things explicit here:

- have the locations in both read and write sets, as these locations we both read and write
- assume VM will later tag correctly, without any "internal additional accesses". i.e. if write does both read and write, VM static analysis will give out both read and write for that location
- total APT supply should then be just a write
- write to a auid generated location (i.e. any object created via Framework) can be a special kind of write (write<new_location) ), as the cryptographic properties guarantee no conflict.